main.tf

```diff
@@ -19,21 +19,20 @@ resource "google_billing_budget" "budget" {
 
   amount {
     dynamic "specified_amount" {
-      for_each = !var.use_last_period_amount ? [true] : []
+      for_each = length(try(var.amount.specified_amount, {})) != 0 ? ["specified_amount"] : []
```

Comment:
@mariux i am sure there is a nicer approach to this, what do you think?

Reply:
please check the how to docs. this is covered in there.